From 143a21cf1789ce673aee9ec7e0ee9110ec38f879 Mon Sep 17 00:00:00 2001
From: Lukas Venhoda <lvenhoda@redhat.com>
Date: Sat, 2 Jan 2016 17:07:52 +0100
Subject: [PATCH] virt-viewer-display-vnc: Set guest name when using VNC

If it's not already set, set guest name field in virt-viewer-app when using VNC.

Wait for VNC to be initialized (virt_viewer_display_vnc_initialized()).
In this callback get field guest name from app and check whether it
was already set before (FE from libvirt).
If not, set the guest name to name provided by VNC from
vnc_display_get_name().

This fill fix issue in remote-viewer: Guest name is Unknown when using VNC.

(cherry picked from commit 772698a8a6e34c0b5051a3519f9284313426c1ca)
---
 src/virt-viewer-display-vnc.c | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/src/virt-viewer-display-vnc.c b/src/virt-viewer-display-vnc.c
index 4b7243b..f477354 100644
--- a/src/virt-viewer-display-vnc.c
+++ b/src/virt-viewer-display-vnc.c
@@ -116,8 +116,22 @@ static void
 virt_viewer_display_vnc_initialized(VncDisplay *vnc G_GNUC_UNUSED,
                                     VirtViewerDisplay *display)
 {
+    gchar *name = NULL;
+    VirtViewerSession *session = virt_viewer_display_get_session(display);
+    VirtViewerApp *app = virt_viewer_session_get_app(session);
+
+    g_object_get(app, "guest-name", &name, NULL);
+    if (name == NULL || *name == '\0') {
+        const gchar * vnc_name = vnc_display_get_name(vnc);
+        if (vnc_name != NULL) {
+            g_object_set(app, "guest-name", vnc_name, NULL);
+        }
+    }
+
     virt_viewer_display_set_show_hint(display,
                                       VIRT_VIEWER_DISPLAY_SHOW_HINT_READY, TRUE);
+
+    g_free(name);
 }
 
 static void
